### ISSUE TITLE
Issue on Debian: badly formatted changelog

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#488]
**Reported by valleegr on 23 Jul 2008 20:10 UTC**
With trunk, when i try to use OPKGC, i have the following error: 

```
$ opkgc -o /tmp --dist=debian
[Read config file /etc/opkgc.conf
[INFO](INFO]) Validating XML file against schema
[opkg tarball created: /tmp/opkg-sis-4.0.0.tar.gz
[INFO](INFO]) Generates /tmp/opkg-sis-4.0.0/debian/copyright from template /usr/local/share/opkgc/tmpl/debian/copyright.tmpl
[Generates /tmp/opkg-sis-4.0.0/debian/README.debian from template /usr/local/share/opkgc/tmpl/debian/README.debian.tmpl
[INFO](INFO]) Generates /tmp/opkg-sis-4.0.0/debian/changelog from template /usr/local/share/opkgc/tmpl/debian/changelog.tmpl
[Copy /usr/local/share/opkgc/tmpl/debian/compat to /tmp/opkg-sis-4.0.0/debian
[INFO](INFO]) Generates /tmp/opkg-sis-4.0.0/debian/control from template /usr/local/share/opkgc/tmpl/debian/control.tmpl
[Copy /usr/local/share/opkgc/tmpl/debian/rules to /tmp/opkg-sis-4.0.0/debian
[INFO](INFO]) parsechangelog/debian: error: badly formatted trailer line, at file debian/changelog line 11
[dpkg-buildpackage: unable to determine source package is
[ERROR](INFO]) Packages generation failed
```

The problem is that the changelog file is not correctly created, for instance, i have the following generated changelog:

```
opkg-sis (4.0.0-3) unstable; urgency=low

  [Kim](DongInn)
  - Add a requirement "xinetd" for ydl ppc
  [Kim](DongInn)
  - Add a requirement "dhcp" for ydl ppc
  [Focht](Erich)
  - Updated systemconfigurator to version 2.2.11-1.
  - Added fixup script for bad systemimager versions doing the rewrite of boot devices.

 --
```

It should be something like:

```
opkg-sis (4.0.0-3) unstable; urgency=low

  [Kim](DongInn)
  - Add a requirement "xinetd" for ydl ppc
  [Kim](DongInn)
  - Add a requirement "dhcp" for ydl ppc
  [Focht](Erich)
  - Updated systemconfigurator to version 2.2.11-1.
  - Added fixup script for bad systemimager versions doing the rewrite of boot devices.

 -- Geoffroy Vallee <valleegr@ornl.gov>  Wed, 23 Jul 2008 11:44:04 -0400
```

**Comment by dikim33** [comment]
**Comment by valleegr on 31 Jul 2008 18:35 UTC**
The following patch is not perfect but allows me to use OPKGC. I also attached the patch as a file.

```
Index: src/Opkgc/Compiler.py
===================================================================
--- src/Opkgc/Compiler.py       (revision 7156)
+++ src/Opkgc/Compiler.py       (working copy)
@@ -162,7 +162,7 @@
     dest_dir = None
     opkgName = None
     configSection = "DEBIAN"
-    buildCmd = "dpkg-buildpackage"
+    buildCmd = "dpkg-buildpackage -uc -us"
     supportedDist = [    def __init__(self, opkgDesc, dest_dir, dist):
Index: src/Opkgc/Deb.py
===================================================================
--- src/Opkgc/Deb.py    (revision 7156)
+++ src/Opkgc/Deb.py    (working copy)
@@ -118,7 +118,8 @@
         name = centry['name']('debian']

)
         date = centry[        validAuthor = False
-        ret = "%s  %s" % (name, self.date(date, "RFC822"))
+        ret = "%s <oscar-devel@lists.sourceforge.net>  %s" % (name, self.date(date, "RFC822"))
+        return ret

     def getSourceFiles(self):
         return [ DebSourceFile(f) for f in self.opkgDesc.getSourceFiles()]('date']
)
Index: templates/debian/changelog.tmpl
===================================================================
--- templates/debian/changelog.tmpl     (revision 7156)
+++ templates/debian/changelog.tmpl     (working copy)
@@ -11,6 +11,6 @@
 #end for
 #end for

- -- $uploader(versionEntry)
+ -- ${uploader($versionEntry)}

 #end for
```

Note this is not an acceptable long-term solution since:
- the email included into the Debian ChangeLog (mandatory) is the oscar-devel mailing list email, which creates problem if we try to automatically sign the created packages,
- because of the signature issue described previously, the patch disable package signing.

**Comment by dikim33** [comment]
**Comment by valleegr on 4 Dec 2008 07:05 UTC**
(In [7638]) This check-in fixes #489 - OPKGC generates now a valid changelog for Debian
packages.
